actions/microsoft-onedrive/microsoft_onedrive/onedrive_get_action.py

```diff
@@ -0,0 +1,331 @@
+
+        if search_request.item_type == "file":
+            filtered_items = [item for item in all_items if "file" in item]
+        elif search_request.item_type == "folder":
+            filtered_items = [item for item in all_items if "folder" in item]
+        else:
+            filtered_items = all_items
+
+        for item in filtered_items:
+            if "file" in item:
+                item_id = item["id"]
+                download_url = f"{BASE_GRAPH_URL}/me/drive/items/{item_id}"
+                download_response = requests.get(download_url, headers=headers)
+
+                if download_response.status_code == 200:
+                    download_info = download_response.json()
+                    item["download_url"] = download_info.get(
+                        "@microsoft.graph.downloadUrl"
+                    )
+                else:
+                    item["download_url"] = None
+            else:
+                item["download_url"] = None
+
+        return Response(result={"items": filtered_items})
+    else:
+        raise ActionError(f"Failed to search items: {response.text}")
+
+
+@action
+def download_onedrive_file(
+    token: OAuth2Secret[
+        Literal["microsoft"],
+        List[Literal["Files.Read.All"]],
+    ],
+    download_request: DownloadRequest,
+    download_location: str = "",
+    download_all: bool = False,
+) -> Response[str]:
+    """
+    Download file or files from OneDrive to local storage.
+
+    Do not give 'download_location' unless user specifies a path.
+    By default downloads to the user's Downloads folder.
+
+    Args:
+        token: OAuth2 token to use for the operation.
+        download_request: The download request containing the search query or download URL.
+        download_location:  The path (folder) to save the downloaded file.
+        download_all: Whether to download all files found or not.
+
+    Returns:
+        Response with a message indicating the number of files downloaded.
+    """
+    download_location = (
+        Path.home() / "Downloads"
+        if download_location == ""
+        else Path(download_location)
+    )
+    if download_request.download_url and download_request.name:
+        matching_files = [
+            {
+                "name": download_request.name,
+                "download_url": download_request.download_url,
+            }
+        ]
+    elif download_request.search_items_request:
+        response = search_for_onedrive_items(
+            token,
+            download_request.search_items_request,
+        )
+        matching_files = response.result.get("items", [])
+        if len(matching_files) == 0:
+            raise ActionError("No files found matching the search query.")
+        elif not download_all and len(matching_files) > 1:
+            raise ActionError(
+                "Multiple files found matching the search query. Set download_all to True to download all files."
+            )
+
```

Comment:
This looks a bit weird, could we remove thsi space and also add a new space after the else to make it easier to read and more organised?